docs/source/frameworks/pytorch.rst

```diff
@@ -2,55 +2,158 @@
 PyTorch
 =======
 
-Here's an example using PyTorch with BentoML:
+BentoML provides native support for serving and deploying models trained from PyTorch.
 
-.. code:: python
+Preface
+-------
+
+User may also convert PyTorch models to torchscript models, and use ``bentoml.torchscript`` 
+to save them. It will be a more stable way to serve and deploy PyTorch models in production.
+
+.. note::
+
+    :bdg-info:`Remarks:` We recommend users apply model optimization techniques such as **distillation** or **quantization**.
+    Alternatively, PyTorch models can also be converted to :ref:`ONNX <frameworks/onnx>` models and leverage different runtimes (e.g. TensorRT, Apache TVM, etc.).
+
+
+Saving a Trained Model
+----------------------
+
+For common PyTorch models with single input:
+
+.. code-block:: python
+    :caption: `train.py`
 
-    import pandas as pd
     import torch
-    import bentoml
+    import torch.nn as nn
+    import torch.nn.functional as F
+    import torch.optim as optim
+
+
+    class Net(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.conv1 = nn.Conv2d(3, 6, 5)
+            self.pool = nn.MaxPool2d(2, 2)
+            self.conv2 = nn.Conv2d(6, 16, 5)
+            self.fc1 = nn.Linear(16 * 5 * 5, 120)
+            self.fc2 = nn.Linear(120, 84)
+            self.fc3 = nn.Linear(84, 10)
 
-    class NGramLanguageModeler(nn.Module):
+        def forward(self, x):
+            x = self.pool(F.relu(self.conv1(x)))
+            x = self.pool(F.relu(self.conv2(x)))
+            x = x.view(-1, 16 * 5 * 5)
+            x = F.relu(self.fc1(x))
+            x = F.relu(self.fc2(x))
+            x = self.fc3(x)
+            return x
 
-    def __init__(self, vocab_size, embedding_dim, context_size):
-        super(NGramLanguageModeler, self).__init__()
-        self.embeddings = nn.Embedding(vocab_size, embedding_dim)
-        self.linear1 = nn.Linear(context_size * embedding_dim, 128)
-        self.linear2 = nn.Linear(128, vocab_size)
 
-    def forward(self, inputs):
-        embeds = self.embeddings(inputs).view((1, -1))
-        out = F.relu(self.linear1(embeds))
-        out = self.linear2(out)
-        log_probs = F.log_softmax(out, dim=1)
-        return log_probs
+    model = Net()
+    ... # training
```

Comment:
add reproducible training code here.

also add a link to the prior tutorial if applicable.